tensorrt_llm/_torch/modules/mamba/mamba2_mixer.py

```diff
@@ -60,7 +60,7 @@ def __init__(
         tp_rank = config.mapping.tp_rank
         tp_size = config.mapping.tp_size
 
-        d_inner = d_model * expand
+        d_inner = head_dim * num_heads
         nheads = d_inner // head_dim
```

Comment:
if `num_heads` is passed as input, we don'y need to compute it again from d_inner..